testing/decorators.py

```diff
@@ -44,9 +47,52 @@ def _apply_confoverride_to_class(cls, config, priority):
         for __, kw in markers_kwargs:
             deep_update(sphinx_kwargs, kw)
 
+        ################################################################################
+        # Automatically create docs/{conf.py,index.rst} for the test project.          #
+        ################################################################################
+        def _with_docs_dir(self):
```

Comment:
I'd think it would be better to leave this code near the ``_set_app`` definition as it's a test case setup thing rather than closely related to how the configuration is determined.

Reply:
I think I've solved it once and for all!  In `conftest.py` you can define these two key methods

```py
def pytest_runtest_setup(item):
def pytest_runtest_teardown(item, nextitem):
```

With some additional class level attribute saving and extra machinery it appears to all be working as expected.  It got pretty ugly, so I'm going to clean that up tomorrow and then push ;)